src/services/server/ServerService.ts

```diff
@@ -39,6 +41,27 @@ export class ServerService {
     });
   }
 
+  getServer(server: IServerAddress): ng.IPromise<IServer> {
+    let deferred: ng.IDeferred<IServer> = this.$q.defer<IServer>();
+    this.dmrService.readResource({
+      address: this.generateAddress(server),
+      "include-runtime": true
+    }).then(server => {
```

Comment:
What happens if the server is not active here? We should check the server status before proceeding and take required action if it stopped.